tools/integration_tests/stale_handle/stale_file_handle_synced_file_test.go

```diff
@@ -0,0 +1,214 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package stale_handle
+
+import (
+	"log"
+	"path"
+	"testing"
+
+	"cloud.google.com/go/storage"
+	. "github.com/googlecloudplatform/gcsfuse/v2/tools/integration_tests/util/client"
+	"github.com/googlecloudplatform/gcsfuse/v2/tools/integration_tests/util/operations"
+	"github.com/googlecloudplatform/gcsfuse/v2/tools/integration_tests/util/setup"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+// //////////////////////////////////////////////////////////////////////
+// Boilerplate
+// //////////////////////////////////////////////////////////////////////
+
+const Content = "foobar"
+const Content2 = "foobar2"
+
+type staleFileHandleSyncedFile struct {
+	flags []string
+	suite.Suite
+}
+
+func (s *staleFileHandleSyncedFile) SetupSuite() {
+	mountGCSFuseAndSetupTestDir(s.flags, ctx, storageClient, testDirName)
+}
+
+func (s *staleFileHandleSyncedFile) TearDownSuite() {
+	setup.UnmountGCSFuse(rootDir)
+}
+
+////////////////////////////////////////////////////////////////////////
+// Tests
+////////////////////////////////////////////////////////////////////////
+
+func (s *staleFileHandleSyncedFile) TestSyncedObjectClobberedRemotelyReadThrowsStaleFileHandleError() {
+	testCaseDir := "TestSyncedObjectClobberedRemotelyReadThrowsStaleFileHandleError"
+	targetDir := path.Join(testDirPath, testCaseDir)
+	operations.CreateDirectory(targetDir, s.T())
+	// Create an object on bucket
+	err := CreateObjectOnGCS(ctx, storageClient, path.Join(testDirName, testCaseDir, FileName1), GCSFileContent)
```

Comment:
same comments as local file tests to refactor setup and teardown apply here too. I would move common parts to setup till file handle is opened since it doesn't matter if we open the file handle in read only, read write or write only mode.